app/src/main/java/ch/epfl/sdp/database/firebase/api/SoloFirestoreDatabaseAPI.java

```diff
@@ -0,0 +1,13 @@
+package ch.epfl.sdp.database.firebase.api;
+
+import com.google.firebase.firestore.FirebaseFirestore;
+import ch.epfl.sdp.entity.PlayerManager;
+
+public class SoloFirestoreDatabaseAPI implements SoloDatabaseAPI {
+    @Override
+    public void updateCurrentUserScore(String currentUserEmail, int generalGameScore) {
+        FirebaseFirestore.getInstance().collection(PlayerManager.USER_COLLECTION_NAME)
+                .document(currentUserEmail)
+                .update("generalScore", generalGameScore);
+    }
```

Comment:
If the player is playing solo (against bots) his general score should not be affected. It's only when there are other players that general score makes sense since then you can rank them like you did with leaderboard. Also since we can't test this, minimizing the surface of firebase in the codebase actually makes increasing coverage easier

Reply:
Now we have the In-game score and General Score, the in-game score is game round-based while the general score is a long-term score which is the sum of the in-game score, which means the more game round you play the higher the general score will be. So no matter the solo mode or the multiplayer mode, the general score will be affected.

Reply:
And the general score leaderboard doesn't mean compare with players in the one game round, it means compare with all the players who use our app.